protocols/gossipsub/src/behaviour.rs

```diff
@@ -1518,13 +1576,6 @@ where
             }
         }
 
-        // If we are not validating messages, assume this message is validated
-        // This will allow the message to be gossiped without explicitly calling
-        // `validate_message`.
-        if !self.config.validate_messages() {
-            raw_message.validated = true;
-        }
-
```

Comment:
The `state` of the `raw_message` gets now set within `forward_or_track_promise`.